platform/kube/inject/inject.go

```diff
@@ -190,6 +193,20 @@ func GetInitializerConfig(kube kubernetes.Interface, namespace, injectConfigName
 		return nil, err
 	}
 
+	if c.IncludeNamespaces != nil && c.ExcludeNamespaces != nil {
+		return nil, fmt.Errorf("cannot configure both namespaces and excludeNamespaces")
+	}
+
+	if c.IncludeNamespaces == nil {
+		c.IncludeNamespaces = []string{v1.NamespaceAll}
```

Comment:
would this change the output of describe the configmap?  ` kubectl describe configmap istio-inject  -n istio-system`

Reply:
I found it is odd we don't allow user to set includeNamespaces but we are setting it ourselves... just wanted to make sure it is not surfaced to the user when user views the configmap.

Reply:
`kubectl describe configmap istio-inject -n istio-system` only show resources in the yaml template.

I think that we need to update the document by clarify that if both `namespaces` and `excludeNamespaces` were not set, initializer will inject data to all namespaces.

Comments @linsun @ayj @rshriram

Reply:
`if c.IncludeNamespaces == nil {}` is only needed to make injection work when `excludeNamespaces` is non-empty. You could add `if c.IncludeNamespaces == nil && c.ExcludeNamespaces == nil {}` to error out when user doesn't specify a namespace.

Reply:
done